lib/net/pool.js

```diff
@@ -2012,6 +2123,99 @@ class Pool extends EventEmitter {
     peer.sendCFCheckpt(packet.filterType, packet.stopHash, filterHeaders);
   }
 
+  /**
+   * Handle peer `CFCheckpt` packet.
+   * @method
+   * @private
+   * @param {Peer} peer
+   * @param {CFCheckptPacket} packet
+   */
+
+  async handleCFCheckpt(peer, packet) {
+    if (!this.options.neutrino) {
+      peer.ban();
+      peer.destroy();
+    }
+  }
+
+  /**
+   * Handle peer `CFHeaders` packet.
+   * @method
+   * @private
+   * @param {Peer} peer - Sender.
+   * @param {CFHeadersPacket} packet - Packet to handle.
+   * @returns {void}
+   */
+  async handleCFHeaders(peer, packet) {
+    this.logger.info('Received CFHeaders packet from %s', peer.hostname());
+    if (!this.options.neutrino) {
+      peer.ban();
+      peer.destroy();
+      return;
+    }
+
+    const filterType = packet.filterType;
+    assert(filterType === this.getcfheadersFilterType);
```

Comment:
do we normally assert like this in packet handlers? or do we just log, ignore the packet, maybe ban the peer for some things?